lang/lua/lua.py

```diff
@@ -0,0 +1,312 @@
+from talon import Context, Module, actions, settings
+
+mod = Module()
+ctx = Context()
+ctx.matches = r"""
+tag: user.lua
+"""
+
+mod.setting(
+    "lua_version",
+    type=float,
+    default=5.1,
+    desc="The default lua version to use. Dictates certain operators",
+)
+mod.tag("stylua", desc="Tag for stylua linting commands")
+
+ctx.lists["user.code_common_function"] = {
+    "to number": "tonumber",
+    "eye pairs": "ipairs",
+    "print": "print",
+    "print F": "printf",
+    "type": "type",
+    "assert": "assert",
+    "get meta table": "getmetatable",
+    "set meta table": "setmetatable",
+    # io
+    "eye oh write": "io.write",
+    "eye oh read": "io.read",
+    "eye oh open": "io.open",
+    # string
+    "format": "string.format",
+    "string G find": "string.gfind",
+    "string find": "string.strfind",
+    "string len": "string.strlen",
+    "string upper": "string.strupper",
+    "string lower": "string.strlower",
+    "string sub": "string.strsub",
+    "string G sub": "string.gsub",
+    "string match": "string.match",
+    "string G match": "string.gmatch",
+    # table
+    "table unpack": "table.unpack",
+    "table insert": "table.insert",
+    "tabel get N": "table.getn",
+    "tabel sort": "table.sort",
+    # math
+    "math max": "math.max",
+    # json
+    "jason parse": "json.parse",
+    # http
+    "web get": "http.get",
+    # os
+    "O S date": "os.date",
+    "O S time": "os.time",
+    "O S clock": "os.clock",
+    "O S rename": "os.rename",
+    "O S remove": "os.remove",
+    "O S getenv": "os.getenv",
+    "O S execute": "os.execute",
+}
+
+ctx.lists["user.code_libraries"] = {
+    "bit": "bit",
+    "I O": "io",
+    "string": "string",
+    "U T F eight": "utf8",
+    "table": "table",
+    "math": "math",
+    "O S": "os",
+    "debug": "debug",
+    "L F S": "lfs",
+    "socket": "socket",
+    "H T T P": "http",
```

Comment:
conformer should be able to handle just "io", "utf eight", "os", "lfs", "http" - are you using Dragon / does Dragon have issues with acronyms?

Reply:
No I use conformer, just an old pre-conformer habit from not wanting to assume stuff isn't known and then not working. Generally (to me) it seems sane to just not assume or spend time verifying what it knows vs not and just spell it out though. Curious if there is an advantage, outside I guess readability, to having utf8 vs U T F 8?

Reply:
IME with conformer splitting words into separate characters actually harms recognition; so "utf eight" will actually recognize more accurately than "u t f eight". But I think it's fine to just have "u t f eight", it may well be more future-proof against different/future models.

Reply:
As is, this will work better for for those that still use dragon too